extmod/modtls_mbedtls.c

```diff
@@ -382,10 +392,87 @@ static mp_obj_t ssl_context_get_ciphers(mp_obj_t self_in) {
 }
 static MP_DEFINE_CONST_FUN_OBJ_1(ssl_context_get_ciphers_obj, ssl_context_get_ciphers);
 
+// Helper function to set PSK ciphersuites
+static void set_psk_ciphersuites(mbedtls_ssl_config *conf) {
+    // Create a list of PSK ciphersuites
+    static int *psk_ciphersuites = NULL;
+
+    if (psk_ciphersuites == NULL) {
+        // Define known PSK ciphersuites
+        // These are common PSK ciphersuites supported by mbedtls
+        static const int known_psk_ciphersuites[] = {
+            MBEDTLS_TLS_PSK_WITH_AES_128_CBC_SHA256,
+            MBEDTLS_TLS_PSK_WITH_AES_128_CBC_SHA,
+            MBEDTLS_TLS_PSK_WITH_AES_256_CBC_SHA,
+            MBEDTLS_TLS_PSK_WITH_AES_128_GCM_SHA256,
+            MBEDTLS_TLS_PSK_WITH_AES_256_GCM_SHA384,
+            0  // Terminating zero
+        };
+        
+        // Count available PSK ciphersuites
+        int count = 0;
+        for (int i = 0; known_psk_ciphersuites[i] != 0; i++) {
+            count++;
+        }
```

Comment:
Get the actual authoritative list of supported PSK suites by filtering the output from `mbedtls_ssl_list_ciphersuites` using `mbedtls_ssl_ciphersuite_from_id` and `mbedtls_ssl_ciphersuite_uses_psk`.